src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1059,12 +1059,22 @@ private static IEnumerable<SelectListItem> GetSelectListItems([NotNull] ViewCont
             values = values.Concat(enumValues);
 
             selectedValues = new HashSet<string>(values, StringComparer.OrdinalIgnoreCase);
+
+            // Perform deep copy of selectList to avoid changing user's Selected property values.
             var newSelectList = new List<SelectListItem>();
             foreach (SelectListItem item in selectList)
             {
-                item.Selected =
+                var selected =
```

Comment:
Maybe null coalesce this for succinctness? 
`var selected = selectedValues.Contains(item.Value ?? item.Text);`